ccd-client/src/test/java/uk/gov/hmcts/cmc/ccd/client/service/StartCaseTest.java

```diff
@@ -0,0 +1,62 @@
+package uk.gov.hmcts.cmc.ccd.client.service;
+
+import com.sun.tools.javadoc.Start;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+import org.mockito.junit.MockitoJUnitRunner;
+import org.springframework.http.*;
+import org.springframework.web.client.RestTemplate;
+import uk.gov.hmcts.cmc.ccd.client.header.HttpHeadersFactory;
+import uk.gov.hmcts.cmc.ccd.client.model.EventRequestData;
+import uk.gov.hmcts.cmc.ccd.client.model.StartEventResponse;
+
+import static org.mockito.Mockito.when;
+import static org.mockito.Mockito.eq;
+
+@RunWith(MockitoJUnitRunner.class)
+public class StartCaseTest {
+
+    @Mock
+    private RestTemplate restTemplate;
+    @Mock
+    private HttpHeadersFactory headersFactory;
+
+    private StartCase startCase;
+
+    private EventRequestData eventRequestData = new EventRequestData();
+
+    @Mock
+    private HttpHeaders httpHeaders;
+
+    @Before
+    public void setup() {
+        MockitoAnnotations.initMocks(this);
+    }
+
+    @Test
+    public void testExchangeReturnsStartEvent() throws Exception {
```

Comment:
That test is mostly useless. It does not tests what event is returned, does not tests right endpoint is called. Any call to `restTemplate` does not make this client work correctly.

I would consider using mocked HTTP service to tests this properly inclusive deserialisation logic.

Reply:
Not ready for review. I've run out icloud space hence not keeping source locally.

Reply:
What are you talking about @singh-surjeet ? That test is part to this pull request and this is WIP pull request so people can give early feedback. I have no idea how you wanna split this into few PR to be honest. When you develop client library I expect it to be well tested, not having some useless tests cases to keep coverage high.

Reply:
Test was committed after PR was raised and never intended to be part of this PR and never claimed that test are finished. I wanted to see code changes and others can look as well so asked him to raise PR WIP. It was never meant to be merged - I approved it for the sake of understanding of work done on it. All reviews will be taken into account and will be worked on and PR will be closed without merging. However, not keen on Feign until multipart file is sorted. Thanks all for spending time on it. Please speak to me if you have any further questions when you're in PF.